--- configure.in	2012-04-01 15:54:51.803000000 -0400
+++ configure.in.new	2012-04-01 15:54:23.178000004 -0400
@@ -20,19 +20,32 @@
    #----------------------------------------------------------------
    # Set the CLASSEXT location
    #----------------------------------------------------------------
+   AC_MSG_CHECKING(if classext directory exists)
    CLASSEXT=$PWD"/../3rdpartypublic/classext"
-   echo "CLASSEXT="$CLASSEXT
+
+   if test -r "$CLASSEXT" ; then
+      AC_MSG_RESULT([$CLASSEXT])
+   else
+      AC_MSG_WARN([not found])
+   fi
 
    #----------------------------------------------------------------
    # Find ij.jar in directory
    #----------------------------------------------------------------
    AC_MSG_CHECKING(for ij.jar)
-   IJJARDIR="$CLASSEXT/ij.jar"
-   echo "IJJARDDIR="$IJJARDIR
-   if test -r "$IJJARDIR" ; then
-      IJJARPATH="$IJJARDIR"
-      AC_MSG_RESULT($IJJARPATH)
-   fi
+   IJJARDIR="$CLASSEXT/ij.jar
+             $IJPATH/ij.jar
+             $IJPATH/lib/ij.jar
+             $IJPATH/ImageJ.app/Contents/Resources/Java/ij.jar
+             $IJPATH/../java/ij.jar"
+
+   for d in $IJJARDIR ; do
+      if test -r "$d" ; then
+         IJJARPATH="$d"
+         AC_MSG_RESULT($IJJARPATH)
+         break
+      fi
+   done
 
    if test -z "$IJJARPATH"; then
      AC_MSG_ERROR([not found, Use the without-imagej directive to build without ImageJ or install ImageJ (http://rsb.info.nih.gov/ij/ and rerun configure, possibly using the --with-imagej=path directive])
@@ -41,16 +54,20 @@
    #----------------------------------------------------------------
    # Find beanshell.jar in ImageJ plugins directory
    #----------------------------------------------------------------
-   AC_MSG_CHECKING(for beanshell jar file in ImageJ plugins directory)
-   if test -r "$CLASSEXT/bsh-2.0b4.jar" ; then
-      BSHJARPATH="$CLASSEXT/bsh-2.0b4.jar"
-      AC_MSG_RESULT($BSHJARPATH)
-   fi
-   if test -r "/usr/share/java/bsh.jar" ; then
-      BSHJARPATH="/usr/share/java/bsh.jar"
-      AC_MSG_RESULT($BSHJARPATH)
-   fi
-   if test "$BSHJARPATH" = "" ; then
+   AC_MSG_CHECKING(for beanshell jar file)
+   BSHJARDIR="$CLASSEXT/bsh-2.0b4.jar
+              /usr/share/java/bsh.jar
+              /usr/share/bsh/lib/bsh.jar"
+
+   for d in $BSHJARDIR ; do
+      if test -r "$d" ; then
+         BSHJARPATH="$d"
+         AC_MSG_RESULT($BSHJARPATH)
+         break
+      fi
+   done
+
+   if test -z "$BSHJARPATH"; then
       AC_MSG_ERROR([not found, Use the without-imagej directive to build without ImageJ GUI, or download Beanshell (http://www.beanshell.org/download.html) and copy bsh-2.0b4.jar to $IJPATH/plugins])
    fi
 
@@ -58,87 +75,120 @@
    # Find swingx.jar in ImageJ plugins directory
    #----------------------------------------------------------------
    AC_MSG_CHECKING(for swingx jar file in ImageJ plugins directory)
-   if test -r "$CLASSEXT/swingx-0.9.5.jar" ; then
-      SWINGXJARPATH="$CLASSEXT/swingx-0.9.5.jar"
-      AC_MSG_RESULT($SWINGXJARPATH)
-   fi
-   if test -r "/usr/share/java/swingx.jar" ; then
-      SWINGXJARPATH="/usr/share/java/swingx.jar"
-      AC_MSG_RESULT($SWINGXJARPATH)
-   fi
-   if test "$SWINGXJARPATH" = "" ; then
+   SWINGXJARDIR="$CLASSEXT/swingx-0.9.5.jar
+                 /usr/share/java/swingx.jar
+                 /usr/share/swingx-1.6/lib/swingx.jar"
+
+   for d in $SWINGXJARDIR ; do
+      if test -r "$d" ; then
+         SWINGXJARPATH="$d"
+         AC_MSG_RESULT($SWINGXJARPATH)
+         break
+      fi
+   done
+
+   if test -z "$SWINGXJARPATH"; then
       AC_MSG_ERROR([not found, Use the without-imagej directive to build without ImageJ GUI, or download swingx and copy to $IJPATH/plugins])
    fi
-   
+
    #----------------------------------------------------------------
    # Find commons-math-2.0.jar in ImageJ plugins directory
    #----------------------------------------------------------------
-
    AC_MSG_CHECKING(for apache commons math jar file)
-   if test -r "$CLASSEXT/commons-math-2.0.jar" ; then
-      COMMONSMATHJARPATH="$CLASSEXT/plugins/commons-math-2.0.jar"
-      AC_MSG_RESULT($COMMONSMATHJARPATH)
-   fi
-   if test -r "/usr/share/java/commons-math-2.0.jar" ; then
-      COMMONSMATHJARPATH="/usr/share/java/commons-math-2.0.jar"
-      AC_MSG_RESULT($COMMONSMATHJARPATH)
-   fi
-   if test -r $CLASSEXT"/commons-math-2.0.jar" ; then
-      COMMONSMATHJARPATH=$CLASSEXT"/commons-math-2.0.jar"
-      AC_MSG_RESULT($COMMONSMATHJARPATH)
-   fi
-   if test "$COMMONSMATHJARPATH" = "" ; then
-      AC_MSG_ERROR([commons-math-2.0.jar not found. It should be in the classext directory.])
+   COMMONSMATHJARDIR="$CLASSEXT/commons-math-2.0.jar
+                      /usr/share/java/commons-math-2.0.jar
+                      /usr/share/commons-math-2/lib/commons-math.jar"
+
+   for d in $COMMONSMATHJARDIR ; do
+      if test -r "$d" ; then
+         COMMONSMATHJARPATH="$d"
+         AC_MSG_RESULT($COMMONSMATHJARPATH)
+         break
+      fi
+   done
+
+   if test -z "$COMMONSMATHJARPATH" ; then
+      AC_MSG_ERROR([commons-math-2.0.jar not found.])
    fi
 
    #----------------------------------------------------------------
    # Find swing-layout-1.0.4.jar in ImageJ plugins directory
    #----------------------------------------------------------------
    AC_MSG_CHECKING(for swing-layout jar file)
-   if test -r "$CLASSEXT/swing-layout-1.0.4.jar" ; then
-      SWINGLAYOUTJARPATH="$CLASSEXT/swing-layout-1.0.4.jar"
-      AC_MSG_RESULT($SWINGLAYOUTJARPATH)
-   fi
-   if test -r "/usr/share/java/swing-layout-1.0.4.jar" ; then
-      SWINGLAYOUTJARPATH="/usr/share/java/swing-layout-1.0.4.jar"
-      AC_MSG_RESULT($SWINGLAYOUTJARPATH)
-   fi
-   if test -r $CLASSEXT"/swing-layout-1.0.4.jar" ; then
-      SWINGLAYOUTJARPATH=$CLASSEXT"/swing-layout-1.0.4.jar"
-      AC_MSG_RESULT($SWINGLAYOUTJARPATH)
-   fi
-   if test "$SWINGLAYOUTJARPATH" = "" ; then
-      AC_MSG_ERROR([swing-layout-1.0.4.jar not found.  It should be in the classext directory.])
+   SWINGLAYOUTJARDIR="$CLASSEXT/swing-layout-1.0.4.jar
+                       /usr/share/java/swing-layout-1.0.4.jar
+                       /usr/share/swing-layout-1/lib/swing-layout.jar"
+
+   for d in $SWINGLAYOUTJARDIR ; do
+      if test -r "$d" ; then
+         SWINGLAYOUTJARPATH="$d"
+         AC_MSG_RESULT($SWINGLAYOUTJARPATH)
+         break
+      fi
+   done
+
+   if test -z "$SWINGLAYOUTJARPATH"; then
+      AC_MSG_ERROR([swing-layout-1.0.4.jar not found.])
    fi
-   
+
    #----------------------------------------------------------------
    # Find AbsoluteLayout.jar 
    #----------------------------------------------------------------
    AC_MSG_CHECKING(for AbsoluteLayout jar file)
-   if test -r "$CLASSEXT/AbsoluteLayout.jar" ; then
-      ABSOLUTEJARPATH="$CLASSEXT/AbsoluteLayout.jar"
-      AC_MSG_RESULT($ABSOLUTEJARPATH)
-   fi
-   if test -r "/usr/share/java/AbsoluteLayout.jar" ; then
-      ABSOLUTEJARPATH="/usr/share/java/AbsoluteLayout.jar"
-      AC_MSG_RESULT($ABOSLUTEJARPATH)
-   fi
-   if test -r $CLASSEXT"/AbsoluteLayout.jar" ; then
-      ABSOLUTEJARPATH=$CLASSEXT"/AbsoluteLayout.jar"
-      AC_MSG_RESULT($ABSOLUTEJARPATH)
-   fi
-   if test "$ABSOLUTEJARPATH" = "" ; then
-      AC_MSG_ERROR([AbsoluteLayout.jar not found.  It should be in the classext directory.])
+   ABSOLUTEJARDIR="$CLASSEXT/AbsoluteLayout.jar
+                   /usr/share/java/AbsoluteLayout.jar
+                   /usr/share/absolutelayout/lib/absolutelayout.jar"
+
+   for d in $ABSOLUTEJARDIR ; do
+      if test -r "$d" ; then
+         ABSOLUTEJARPATH="$d"
+         AC_MSG_RESULT($ABSOLUTEJARPATH)
+         break
+      fi
+   done
+
+   if test -z "$ABSOLUTEJARPATH"; then
+      AC_MSG_ERROR([AbsoluteLayout.jar not found.])
    fi
+
    #----------------------------------------------------------------
    # Find JFreeChart 
    #----------------------------------------------------------------
    AC_MSG_CHECKING(for JFreeChart jar file)
-   if test -r "$CLASSEXT/jfreechart-1.0.13.jar" ; then
-      JFREECHARTJARPATH="$CLASSEXT/jfreechart-1.0.13.jar:$CLASSEXT/jcommon-1.0.16.jar"
-      AC_MSG_RESULT($JFREECHARTJARPATH)
+   JFREECHARTJARDIR="$CLASSEXT/jfreechart-1.0.13.jar
+                     /usr/share/java/jfreechart-1.0.13.jar
+                     /usr/share/jfreechart-1.0/lib/jfreechart.jar"
+   
+   for d in $JFREECHARTJARDIR ; do
+      if test -r "$d" ; then
+         JFREECHARTJARPATH="$d"
+         AC_MSG_RESULT($JFREECHARTJARPATH)
+         break
+      fi
+   done
+
+   if test -z "$JFREECHARTJARPATH"; then
+      AC_MSG_ERROR([jfreechart-1.0.13.jar not found.])
+   fi
+
+   #----------------------------------------------------------------
+   # Find clojure.jar
+   #----------------------------------------------------------------
+   AC_MSG_CHECKING(for clojure jar file)
+   CLOJUREJARDIR="$CLASSEXT/clojure.jar
+                  /usr/share/clojure-1.3/lib/clojure.jar"
+   
+   for d in $CLOJUREJARDIR ; do
+      if test -r "$d" ; then
+         CLOJUREJARPATH="$d"
+         AC_MSG_RESULT($CLOJUREJARPATH)
+         break
+      fi
+   done
+
+   if test -z "$CLOJUREJARPATH"; then
+      AC_MSG_ERROR([clojure.jar not found.])
    fi
-   # TODO: look in other paths
 fi
 
 
@@ -149,6 +199,7 @@
 AC_SUBST(SWINGLAYOUTJARPATH)
 AC_SUBST(ABSOLUTEJARPATH)
 AC_SUBST(JFREECHARTJARPATH)
+AC_SUBST(CLOJUREJARPATH)
 AC_SUBST(IJJARPATH)
 
 AC_MSG_CHECKING([whether to build the core])
